src/analyses/invariant_set.cpp

```diff
@@ -569,12 +568,13 @@ void invariant_sett::strengthen_rec(const exprt &expr)
   }
   else if(expr.id()==ID_notequal)
   {
-    assert(expr.operands().size()==2);
+    const auto &notequal_expr = to_notequal_expr(expr);
 
     std::pair<unsigned, unsigned> p;
 
-    if(get_object(expr.op0(), p.first) ||
-       get_object(expr.op1(), p.second))
+    if(
+      get_object(notequal_expr.op0(), p.first) ||
+      get_object(notequal_expr.op1(), p.second))
```

Comment:
Nit pick: please add braces for readability.

Reply:
done